src/value_log.rs

```diff
@@ -270,7 +282,17 @@ impl<BC: BlobCache, C: Compressor + Clone> ValueLog<BC, C> {
             return Ok(None);
         };
 
-        let mut reader = BufReader::new(File::open(&segment.path)?);
+        let fd = match self.fd_cache.get(self.id, vhandle.segment_id) {
+            Some(fd) => fd,
+            None => {
+                let fd = File::open(&segment.path)?;
+                let fd_clone = fd.try_clone()?;
+                self.fd_cache.insert(self.id, vhandle.segment_id, fd_clone);
```

Comment:
Why clone the FD here - instead it could be first used, then moved into the cache.

Reply:
For this case, because File's ownership is moved to BufReader, so we can't insert it to the cache after its used.

Even if we were to cache BufReader instead, we would still have a similar problem because of SegmentReader

Am I missing something here?

Reply:
`BufReader::into_inner` allows destroying the `BufReader` and get its internal T. However each BufReader construction is still a heap allocation (https://github.com/rust-lang/rust/blob/e7df5b055d1ea724b225c44bd5084141d6ab42b4/library/std/src/io/buffered/bufreader/buffer.rs#L35), so it's a bit wasteful to create, then destroy it every time. For now it would be best to just store a `BufReader` instead.

V3 will probably change in such way to `BufReader` is not needed anymore though.